File: number-of-connected-componets-in-undirected-graph/numberOfConnectedComponents.py
class Solution(object):
    def countComponents(self, n, edges):
        """
        :type n: int
        :type edges: List[List[int]]
        :rtype: int
        """
        path = [i for i in range(n)]
        for edge in edges:
            a = self.find(path, edge[0])
            b = self.find(path, edge[1])
            if a != b:
                path[a] = b
                n -= 1
        return n
        
    def find(self, path, value):
        while path[value] != value:
            path[value] = path[path[value]]
            value = path[value]
        return value

File: number-of-connected-componets-in-undirected-graph/numberOfConnectedComponents.java
public class Solution {
    public int countComponents(int n, int[][] edges) {
        int[] set = new int[n];
        for(int i = 0; i < n; ++i) {
            set[i] = i;
        }
        for(int[] edge : edges) {
            int a = find(set, edge[0]);
            int b = find(set, edge[1]);
            if(a != b) {
                set[a] = b;
                n--;
            }
        }
        return n;
    }
    
    public int find(int[] set, int id) {
        while(set[id] != id) {
            set[id] = set[set[id]];
            id = set[id];
        }
        return id;
    }
}